vpr/src/device/rr_graph_builder.h

```diff
@@ -40,6 +40,37 @@ class RRGraphBuilder {
     t_rr_graph_storage& node_storage();
     /** @brief Return a writable object for update the fast look-up of rr_node */
     RRSpatialLookup& node_lookup();
+
+    /** @brief Add a rr_segment to the routing resource graph. Return an valid id if successful.*/
```

Comment:
As suggested by @vaughnbetz , we need to provide more intuitive comments to guide developers

```
/** @brief Add a rr_segment to the routing resource graph. Return an valid id if successful.
  *  Each rr_segment contains the detailed information of a routing track, which is denoted by a node in CHANX or CHANY type. It is frequently used by client functions in timing and routability prediction
  */
```